### ISSUE TITLE
Constellation plots: calculating knn in reduced dimension space - pca or umap?

### ISSUE THREAD

**Issue by carmensandoval** [AllenInstitute/scrattch.hicat#58]
Hi scrattch.hicat team,

I've been trying  make some constellation plots of my own using the code in your package.

In your methods section for the Yao 2020 preprint, you describe the process for making the constellation plots:

> For each cell its 15 nearest neighbors in _**reduced dimension space**_ were determined and summarized by cluster. For each cluster, we then calculated the fraction of nearest neighbors that were assigned to other clusters.

Does "reduced dimension space" here refer to PCA, or UMAP? 
And if PCA - how many PCs did you use?

I understand that the **_cluster nodes_** are derived from the UMAP coordinates (centroids), but it's not clear from the explanation or the code if you are getting the knn table from PCA or UMAP coordinates. My hunch is that you use PCA for this, following the workflow used for clustering. Am I right about this?

Thanks a lot!
Carmen

**Comment by yzizhen** [comment]
Yes, you are right that we refer to the PCA space, not UMAP. UMAP is only used for visualization. We used 50-60PCs, but it depends on the complicity of the dataset. Elbow test or other z-score based test can be used for decision.

**Comment by salwanbutrus** [comment]
Hey Carmen!

Did you end successfully generating a constellation plot? I have been having issues creating the correct input for the get_knn_graph() function and was wondering if you had any example csv files laying around. I posted a separate issue on this github, but figured I should ask here as well since this repository doesn't seem to be closely monitored. 

Thanks,
Salwan

**Comment by chhatbarchintan** [comment]
@salwanbutrus did you solve the problem with regards to correct input for the get_knn_graph() function?

**Comment by salwanbutrus** [comment]
I did not. Have you?

**Comment by cvanvelt** [comment]
[This website](https://alleninstitute.github.io/scrattch.example/Constellation.html ) is a work in progress but there is an example for plotting a constellation.
